### ISSUE TITLE
Accordion accessibility fixes

### ISSUE THREAD

**Issue by pyrello** [uiowa/uids#961]
# Motivation

When @bspeare was testing #955 he identified some issue using some browser a11y testing tools.

From AInspector:
![Screenshot 2024-12-20 at 9 56 43 AM](https://github.com/user-attachments/assets/c8a16669-50d6-454c-b79b-f7530a2dcada)
![Screenshot 2024-12-20 at 9 57 17 AM](https://github.com/user-attachments/assets/17001eb8-b3da-48dc-a0d7-4dabb3a5fd7b)
![Screenshot 2024-12-20 at 9 57 29 AM](https://github.com/user-attachments/assets/d0adcac6-2625-4532-ade4-9b211a7a5a70)

From Chrome Lighthouse:
![Screenshot 2024-12-20 at 10 26 19 AM](https://github.com/user-attachments/assets/a04f4c20-220e-4343-9912-431b2c006ee4)

**Comment by joewhitsitt** [comment]
After looking into this a bit, it is my understanding that our adoption of details/summary makes adding the aria-labelledby, aria-expanded, aria-selected attributes no longer necessary, but holding on to the tablist/tab stuff makes that move fuzzy.

I imagine the good news is that we can clean some stuff up, but the bad news is that we might provide less info to the user in order to appease the accessibility checker.

I am going to do a bit of before/after testing to understand what info, if any, might be lost when using VoiceOver.

**Comment by joewhitsitt** [comment]
Yeah, this is getting to be "damned if you do, damned if you don't"

Here is says what roles can be used on details/summary. In other words, not `tab` and use of `tablist` is expecting children of `tab` https://www.w3.org/TR/html-aria/

That aside, here is what I have so far (JS changes not included)
```diff
diff --git a/src/components/accordion/Accordion.vue b/src/components/accordion/Accordion.vue
index a05f82017c..7a9ea5788e 100644
--- a/src/components/accordion/Accordion.vue
+++ b/src/components/accordion/Accordion.vue
@@ -52,14 +52,12 @@ onMounted(() => {
     <details v-for="(item, index) in props.items"
       :key="index"
       class="accordion__item"
-      :aria-labelledby="'accordion-heading-' + index"
       :name="!props.multiselectable ? 'accordion-collection' : null"
       :open="item.open"
       :id="'accordion-item-' + index"
+      role="none"
     >
       <summary :id="'accordion-heading-' + index" class="accordion__heading"
-        :aria-expanded="item.open ? 'true' : 'false'"
-        :aria-selected="item.open ? 'true' : 'false'"
         role="tab"
       >
         <h2>
@@ -70,7 +68,6 @@ onMounted(() => {
       <div
         :id="'accordion-content-' + index"
         class="accordion__content"
-        :aria-labelledby="'accordion-heading-' + index"
         v-html="item.content"
       ></div>
     </details>
```

- AI Inspector likes the changes and doesn't report any issues related to this part of the DOM.
- Lighthouse/Axe gives it a 99% up from the current 93% but still doesn't like tablist/details paired together.
  - `Uses ARIA roles on incompatible elements - Many HTML elements can only be assigned certain ARIA roles. Using ARIA roles where they are not allowed can interfere with the accessibility of the web page.`
- Wave doesn't seem to have an eye for it and says we good. 🤷 

What I am noticing in VoiceOver is that the latest version nor my current version announce that I have toggled the details. Hitting enter is silent. Using the keyboard combination that VoiceOver suggests does register a small beep for both versions. The grouping information information remains.

Update: I had the preferred key-combo wrong in VoiceOver. When using the `ctrl + option + space` combo, I was getting selected/expanded info.

I also found this
https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/tablist_role - "When creating a multi-selectable tablist, include aria-multiselectable="true" on the tablist element."
